src/Ed25519VerificationKey2018.js

```diff
@@ -0,0 +1,459 @@
+   * @example
+   * > const verifier = keyPair.verifier();
+   * > verifier
+   * { verify: [AsyncFunction: verify] }
+   * > verifier.verify(key);
+   *
+   * @returns {{verify: Function}} Used to verify jsonld-signatures.
+   */
+  /* eslint-enable */
+  verifier() {
+    return ed25519VerifierFactory(this);
+  }
+
+  /* eslint-disable max-len */
+  /**
+   * Adds a public key base to a public key node.
+   * @example
+   * > keyPair.addEncodedPublicKey({});
+   * { publicKeyBase58: 'GycSSui454dpYRKiFdsQ5uaE8Gy3ac6dSMPcAoQsk8yq' }
+   * @param {Object} publicKeyNode - The public key node in a jsonld-signature.
+   * @param {string} publicKeyNode.publicKeyBase58 - Base58 Public Key for
+   * [jsonld-signatures]{@link https://github.com/digitalbazaar/jsonld-signatures}.
+   *
+   * @returns {{verify: Function}} A PublicKeyNode in a block.
+   */
+  /* eslint-enable */
+  addPublicKey(publicKeyNode) {
+    publicKeyNode.publicKeyBase58 = this.publicKeyBase58;
+    return publicKeyNode;
+  }
+
+  /**
+   * Adds an encrypted private key to the KeyPair.
+   * @param {Object} keyNode - A plain object.
+   *
+   * @return {Object} The keyNode with an encrypted private key attached.
+   */
+  async addPrivateKey(keyNode) {
+    keyNode.privateKeyBase58 = this.privateKeyBase58;
+    return keyNode;
+  }
+
+  /**
+   * Generates and returns a multiformats encoded
+   * ed25519 public key fingerprint (for use with cryptonyms, for example).
+   * @see https://github.com/multiformats/multicodec
+   *
+   * @param {string} publicKeyBase58 - The base58 encoded public key material.
+   *
+   * @returns {string} The fingerprint.
+   */
+  static fingerprintFromPublicKey({publicKeyBase58}) {
+    // ed25519 cryptonyms are multicodec encoded values, specifically:
+    // (multicodec ed25519-pub 0xed01 + key bytes)
+    const pubkeyBytes = util.base58Decode({
+      decode: base58.decode,
+      keyMaterial: publicKeyBase58,
+      type: 'public'
+    });
+    const buffer = new Uint8Array(2 + pubkeyBytes.length);
+    buffer[0] = 0xed;
+    buffer[1] = 0x01;
+    buffer.set(pubkeyBytes, 2);
+    // prefix with `z` to indicate multi-base base58btc encoding
+    return `z${base58.encode(buffer)}`;
+  }
+
+  /**
+   * Generates and returns a multiformats encoded
+   * ed25519 public key fingerprint (for use with cryptonyms, for example).
+   * @see https://github.com/multiformats/multicodec
+   *
+   * @returns {string} The fingerprint.
+   */
+  fingerprint() {
+    const {publicKeyBase58} = this;
+    return Ed25519VerificationKey2018
+      .fingerprintFromPublicKey({publicKeyBase58});
+  }
+
+  /**
+   * Tests whether the fingerprint was
+   * generated from a given key pair.
+   * @example
+   * > edKeyPair.verifyFingerprint('z2S2Q6MkaFJewa');
+   * {valid: true};
+   * @param {string} fingerprint - A Base58 public key.
+   *
+   * @returns {Object} An object indicating valid is true or false.
+   */
+  verifyFingerprint(fingerprint) {
```

Comment:
Why doesn't this use named params like the other functions? Looking back, there seems to be quite a mix of both styles. Can we just use named params consistently instead?

Reply:
Good point, will fix.

Reply:
Fixed, now uses named params.